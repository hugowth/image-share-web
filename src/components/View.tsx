import React from "react";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() => ({
  root: {
    maxWidth: 650,
    margin: "2rem auto 3rem",
  },
}));

type Props = {
  title: string;
  children: React.ReactNode;
};

const View = ({ title, children }: Props) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <h1>{title}</h1>
      {children}
    </div>
  );
};

export default View;
