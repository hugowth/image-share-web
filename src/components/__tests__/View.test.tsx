import React from "react";
import renderer from "react-test-renderer";

import View from "../View";

describe("View", () => {
  it("renders correctly", () => {
    const title = "title";

    const tree = renderer
      .create(
        <View title={title}>
          <></>
        </View>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
