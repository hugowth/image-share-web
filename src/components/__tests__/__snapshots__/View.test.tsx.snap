// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <h1>
    title
  </h1>
</div>
`;
